### DISCUSSION TITLE
Does authentication work with existing solutions or is always a separate Kratos instance needed?

### DISCUSSION THREAD

**Discussion by alexanderadam** [pixeleye-io/pixeleye#267] [Q&A]
Hi folks,

pixeleye looks amazing. Does authentication work with existing and standardized OIDC/OAuth2 authentication solutions like [KanIDM](https://kanidm.com/) or is always a separate Kratos instance needed?

I saw that GitHub authentication is allowed too and the parameters look pretty much like generic OIDC parameters (i.e. [`APP_NAME`, `APP_CLIENT_ID`, `APP_ID`, `APP_CLIENT_SECRET`, `APP_PRIVATE_KEY`](https://github.com/pixeleye-io/pixeleye/blob/c4e57586a14854aa49cea37acdec9707284aa584/env.template#L33-L38)).

So probably only proper URLs would be missing, right? :thinking: 

In other applications (i.e. [Penpot](https://help.penpot.app/technical-guide/configuration/#openid-connect), [Vikunja](https://vikunja.io/docs/openid-example-configurations/) or [Tolgee](https://tolgee.io/platform/self_hosting/configuration?config-format=env#oauth2)) there's a generic way of defining these variables — only the auth URL and logout URLs are different.

**Comment by AlfieJones** [comment]
Hey, sorry for the late respones, I didn't have proper notifications setup for discussions.

We require a separate kratos instance but that does support generic OIDC so it should be fairly trivial to expand out supporting other login methods.

I'd have to experiment with it but I'll see if I can get it working with others.

I hope this helps
